Realm/RLMCollection.mm

```diff
@@ -468,18 +469,22 @@ - (void)invalidate {
 template<typename RLMCollection>
 RLMNotificationToken *RLMAddNotificationBlock(RLMCollection *collection,
                                               void (^block)(id, RLMCollectionChange *, NSError *),
+                                              NSArray<NSString *> *keyPaths,
                                               dispatch_queue_t queue) {
     RLMRealm *realm = collection.realm;
     if (!realm) {
         @throw RLMException(@"Linking objects notifications are only supported on managed objects.");
     }
     bool skipFirst = std::is_same_v<RLMCollection, RLMResults>;
     auto token = [[RLMCancellationToken alloc] init];
+    
+    RLMClassInfo *info = collection.objectInfo;
+    realm::KeyPathArray keyPathArray = KeyPathArrayFromStringArray(realm, info, keyPaths);
 
     if (!queue) {
         [realm verifyNotificationsAreSupported:true];
         token->_realm = realm;
-        token->_token = RLMGetBackingCollection(collection).add_notification_callback(CollectionCallbackWrapper{block, collection, skipFirst});
+        token->_token = RLMGetBackingCollection(collection).add_notification_callback(CollectionCallbackWrapper{block, collection, skipFirst}, keyPathArray);
```

Comment:
```suggestion
        token->_token = RLMGetBackingCollection(collection).add_notification_callback(CollectionCallbackWrapper{block, collection, skipFirst}, std::move(keyPathArray));
```

Reply:
For my own understanding --  what effect will the move have?

Reply:
Without the std::move() the array will be deep copied at each step that takes it by value, and with std::move() no copying will occur.